File: Spring/Chapter4/java_demo/demo/src/main/java/com/aspectj_xml/XMLImpl.java
package com.aspectj_xml;

public class XMLImpl implements XMLInterface {
    @Override
    public void save() {
        System.out.println("保存");
    }

    @Override
    public void modify() {
        System.out.println("修改");
    }

    @Override
    public void delete() {
        System.out.println("删除");
//        注释下面这行语句开启异常通知
//        int a = 1/0;
    }
}

File: Spring/Chapter4/kotlin_demo/demo/src/main/kotlin/aspectj_xml/XMLImpl.kt
package aspectj_xml

class XMLImpl : XMLInterface{
    override fun save() {
        println("保存")
    }

    override fun modify() {
        println("修改")
    }

    override fun delete() {
        println("删除")
//        异常通知
//        val a = 1/0
    }
}